diff --git a/coprocessor/features/eh_cp_feat_rpc_ext_v2/src/eh_cp_feat_rpc_ext_v2_handler_req_system.c b/coprocessor/features/eh_cp_feat_rpc_ext_v2/src/eh_cp_feat_rpc_ext_v2_handler_req_system.c
--- a/coprocessor/features/eh_cp_feat_rpc_ext_v2/src/eh_cp_feat_rpc_ext_v2_handler_req_system.c
+++ b/coprocessor/features/eh_cp_feat_rpc_ext_v2/src/eh_cp_feat_rpc_ext_v2_handler_req_system.c
@@ -107,7 +107,7 @@
 	}
 
 	ESP_LOGI(TAG, "Prepare partition for OTA\n");
-	ret = esp_ota_begin(update_partition, OTA_SIZE_UNKNOWN, &handle);
+	ret = esp_ota_begin(update_partition, OTA_WITH_SEQUENTIAL_WRITES, &handle);
 	if (ret) {
 		ESP_LOGE(TAG, "OTA begin failed[%d]", ret);
 		s_ota_status = OTA_FAILED;
